backend/app/notify/email.go

```diff
@@ -56,48 +65,11 @@ func NewEmail(server string, port int, username string, password string, keepAli
 			res.server, res.port, err)
 	}
 
-	go func() {
-		d := gomail.NewDialer(res.server, res.port, res.username, res.password)
-
-		var s gomail.SendCloser
-		var err error
-		open := false
-		for {
-			select {
-			case m, ok := <-res.ch:
-				if !ok {
-					return
-				}
-				if !open {
-					if s, err = d.Dial(); err != nil {
-						log.Printf(
-							"[WARN] error connecting to %s:%d with username %s: %s",
-							res.server, res.port, res.username, err)
-					}
-					open = true
-				}
-				if err := gomail.Send(s, m); err != nil {
-					log.Printf(
-						"[INFO] error sending to %s:%d with username %s: %s",
-						res.server, res.port, res.username, err)
-				}
-			// Close the connection to the SMTP server if no email was sent in the keepAlive period.
-			case <-time.After(res.keepAlive):
-				if open {
-					if err := s.Close(); err != nil {
-						log.Printf(
-							"[WARN] error closing connection with %s:%d with username %s: %s",
-							res.server, res.port, res.username, err)
-					}
-					open = false
-				}
-			}
-		}
-	}()
+	go res.activate()
 
 	// TODO: do we need to close this?
```

Comment:
we do. At least to be able to properly test this. We can extend `Destination` interface with Close method and delegate the call to notify.Close